controllers/controllers/utils/handlerutil/capiobjects.go

```diff
@@ -0,0 +1,41 @@
+package handlerutil
+
+import (
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/aws/eks-anywhere/pkg/clusterapi"
+)
+
+// CAPIObjectToCluster returns a request handler that enqueues an EKS-A Cluster
+// reconcile request for CAPI objects that contain the cluster name and namespace labels
+func CAPIObjectToCluster(log logr.Logger) handler.MapFunc {
+	return func(o client.Object) []reconcile.Request {
+		labels := o.GetLabels()
+		clusterName, ok := labels[clusterapi.EKSAClusterLabelName]
+		if !ok {
+			// Object not managed by a eks-a Cluster, don't enqueue
+			log.V(6).Info("Object not managed by an eks-a Cluster, ignoring", "type", fmt.Sprintf("%T", o), "name", o.GetName())
+			return nil
+		}
+
+		clusterNamespace := labels[clusterapi.EKSAClusterLabelNamespace]
+		if clusterNamespace == "" {
```

Comment:
when will this happen? should we fill the namespace and enque?

Reply:
It should never happen under normal circumstances
But I think the code should be able to handle that scenario since labels are a 'free form" field and they can be updated by other controllers, humans, etc.

I don't think we can fill the namespace since we don't know were the eksa cluster is.